gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -161,6 +161,14 @@ private BoundField createBoundField(
 
     @SuppressWarnings("unchecked")
     final TypeAdapter<Object> typeAdapter = (TypeAdapter<Object>) mapped;
+    final TypeAdapter<Object> writeTypeAdapter;
+    if (serialize) {
+      writeTypeAdapter = jsonAdapterPresent ? typeAdapter
+          : new TypeAdapterRuntimeTypeWrapper<>(context, typeAdapter, fieldType.getType());
+    } else {
+      // The field will never be serialized, skip the unnecessary type adapter construction
+      writeTypeAdapter = null;
```

Comment:
While this is correct, I think it's likely to confuse nullness-analysis tools. It's just as easy to set this to `typeAdapter` here, which should confuse them less.

Reply:
Fair point, updated now.